planner/core/casetest/partition/testdata/integration_partition_suite_out.json

```diff
@@ -1672,187 +1672,762 @@
         "DynamicPlan": [
-          "Batch_Point_Get 2.00 root table:t1, index:PRIMARY(a, b) keep order:false, desc:false"
+          "Batch_Point_Get 2.00 root table:thash1, index:PRIMARY(a, b) keep order:false, desc:false"
         ],
         "StaticPlan": [
-          "Batch_Point_Get 2.00 root table:t1, index:PRIMARY(a, b) keep order:false, desc:false"
+          "Batch_Point_Get 2.00 root table:thash1, index:PRIMARY(a, b) keep order:false, desc:false"
         ],
         "Result": [
           "1 1",
           "2 1"
         ]
       },
       {
-        "SQL": "select * from t1 where a = 1 and b in (1,2)",
+        "SQL": "select * from thash1 where a in (1,2) and b = 1 order by a",
         "DynamicPlan": [
-          "Batch_Point_Get 2.00 root table:t1, index:PRIMARY(a, b) keep order:false, desc:false"
+          "IndexReader 2.00 root partition:p1 index:IndexRangeScan",
+          "└─IndexRangeScan 2.00 cop[tikv] table:thash1, index:PRIMARY(a, b) range:[1 1,1 1], [2 1,2 1], keep order:true, stats:pseudo"
+        ],
+        "StaticPlan": [
+          "Batch_Point_Get 2.00 root table:thash1, index:PRIMARY(a, b) keep order:true, desc:false"
+        ],
+        "Result": [
+          "1 1",
+          "2 1"
+        ]
+      },
+      {
+        "SQL": "select * from thash1 where a in (1,2) and b = 1 order by a desc",
+        "DynamicPlan": [
+          "IndexReader 2.00 root partition:p1 index:IndexRangeScan",
+          "└─IndexRangeScan 2.00 cop[tikv] table:thash1, index:PRIMARY(a, b) range:[1 1,1 1], [2 1,2 1], keep order:true, desc, stats:pseudo"
+        ],
+        "StaticPlan": [
+          "Batch_Point_Get 2.00 root table:thash1, index:PRIMARY(a, b) keep order:true, desc:true"
+        ],
+        "Result": [
+          "2 1",
+          "1 1"
+        ]
+      },
+      {
+        "SQL": "select * from thash1 where a = 1 and b in (1,2)",
+        "DynamicPlan": [
+          "Batch_Point_Get 2.00 root table:thash1, index:PRIMARY(a, b) keep order:false, desc:false"
         ],
         "StaticPlan": [
           "PartitionUnion 4.00 root  ",
-          "├─Batch_Point_Get 2.00 root table:t1, index:PRIMARY(a, b) keep order:false, desc:false",
-          "└─Batch_Point_Get 2.00 root table:t1, index:PRIMARY(a, b) keep order:false, desc:false"
+          "├─Batch_Point_Get 2.00 root table:thash1, index:PRIMARY(a, b) keep order:false, desc:false",
+          "└─Batch_Point_Get 2.00 root table:thash1, index:PRIMARY(a, b) keep order:false, desc:false"
         ],
         "Result": [
           "1 1",
           "1 2"
         ]
       },
       {
-        "SQL": "select * from t2 where a in (1,2) and b = 1",
+        "SQL": "select * from thash1 where a = 1 and b in (1,2) order by b",
+        "DynamicPlan": [
+          "IndexReader 2.00 root partition:p0,p1 index:IndexRangeScan",
+          "└─IndexRangeScan 2.00 cop[tikv] table:thash1, index:PRIMARY(a, b) range:[1 1,1 1], [1 2,1 2], keep order:true, stats:pseudo"
+        ],
```

Comment:
What is the reason for Dynamic prune mode does not do Batch_Point_Get here? Sorting?

Reply:
Yes, it will get wrong result. The sorting between different partition table will get wrong results.